File: frontend/src/types/db/tag.ts
import { z } from "zod";

const PostTagSchema = z.object({
    post_id: z.number(),
    tag_id: z.number(),
})

type PostTag = z.infer<typeof PostTagSchema>;

const TagSchema = z.object({
    id: z.number(),
    name: z.string(),
    created_at: z.string().datetime(),
    updated_at: z.string().datetime(),
})

type Tag = z.infer<typeof TagSchema>;

export { PostTagSchema, TagSchema };
export type { PostTag, Tag };

File: backend/internal/domain/entity/tag.go
package entity

import (
	"time"
)

type Tag struct {
	ID        int64
	Name      string
	CreatedAt time.Time
	UpdatedAt time.Time
}

func NewTag(name string) *Tag {
	now := time.Now()
	return &Tag{
		Name:      name,
		CreatedAt: now,
		UpdatedAt: now,
	}
}

type PostTag struct {
	ID        int64
	PostID    int64
	TagID     int64
	CreatedAt time.Time
}

func NewPostTag(postID, tagID int64) *PostTag {
	return &PostTag{
		PostID:    postID,
		TagID:     tagID,
		CreatedAt: time.Now(),
	}
}